rules/CodeQuality/Rector/Foreach_/SimplifyForeachToCoalescingRector.php

```diff
@@ -169,6 +170,11 @@ private function matchForeachReturnOrAssign(Foreach_ $foreach): Expression|Retur
             return null;
         }
 
+        $foreachExprType = $this->nodeTypeResolver->getNativeType($foreach->expr);
+        if (! $foreachExprType->isArray()->yes()) {
+            return null;
+        }
```

Comment:
this ensure it only change native array type, as change other type can be any type, eg: `Generator` or object extends `stdClass`, see https://3v4l.org/6i6kj